### DISCUSSION TITLE
Call a function before page leave or wire:navigate

### DISCUSSION THREAD

**Discussion by T3mmyKay** [filamentphp/filament#13065] [Help]
### Package

Panel builder

### Package Version

v3.2.0

### How can we help you?

I need to be able to call a function in the beforeUnload or wire:navigate event also I need to access the properties of createRecord page and pass it to the function. What I am trying to achieve is on mount a task is created on the db, and before page leaving, the task status is updated

**Comment by T3mmyKay** [comment]
@script
    <script>
        document.addEventListener('visibilitychange', function () {
            console.log($wire.task_id);
            if (typeof @this !== 'undefined') {
                try {
                    navigator.sendBeacon(`/tasks/${$wire.task_id}/cancel`);
                } catch (e) {
                    console.log(e);
                }
            }
        });

        document.addEventListener('livewire:navigate', () => {
            console.log('navigating');
            try {
                navigator.sendBeacon(`/tasks/${$wire.task_id}/cancel`);
            } catch (e) {
                console.log(e);
            }
        })


    </script>
    @endscript
    
    
    This was how I was doing it before, I started using filamentphp

**Comment by leandrocfe** [comment]
Maybe `afterCreate()` ?
https://filamentphp.com/docs/3.x/panels/resources/creating-records#lifecycle-hooks

I think `$this->getRecord()` is available

**Reply by T3mmyKay** [reply]
I am trying to capture maybe the user cancelled creating the model by either reloading the page or moving to another page. Like i said on mount a task is created in the db so unmouunt I want to update the task status. Please any help would be appreciated

**Comment by T3mmyKay** [comment]
I solved this by creating a custom view for the create record
